### ISSUE TITLE
Warning: Multiple GoTrueClient instances detected

### ISSUE THREAD

**Issue by rphlmr** [supabase/gotrue-js#725]
# Bug report

- [x] I confirm this is a bug with Supabase, not with my own application. 
- [x] I confirm I have searched the [Docs](https://docs.supabase.com), GitHub [Discussions](https://github.com/supabase/supabase/discussions), and [Discord](https://discord.supabase.com). 

## Describe the bug

```txt
Multiple GoTrueClient instances detected in the same browser context. It is not an error, but this should be avoided as it may produce undefined behavior when used concurrently under the same storage key.
```

Should we worry to see this warning in browser console? (SSR, Remix Run)
It comes from: https://github.com/supabase/gotrue-js/blob/44af61e872fafbbe4a0438f8ad170f0ce5a1acdd/src/GoTrueClient.ts#L164

I have `persistSession: false` in `createClient` auth's options.

## To Reproduce

-

## Expected behavior

-

## Screenshots

-

## System information

- OS:[macOS
- Browser (if applies) [chrome]
- Version of supabase-js: [2.26.0]
- Version of Node.js: [18.13.0]

## Additional context

-

**Comment by rphlmr** [comment]
Sorry for the inconvenience, I solved it by rewriting some parts of my code by creating a single instance of `SupabaseClient`.
Previously I used a function to create this client 'on demand',

**Comment by maaajo** [comment]
@rphlmr I'm having currently the same issue. What was the fix that solved it for you?

**Comment by rphlmr** [comment]
> @rphlmr I'm having currently the same issue. What was the fix that solved it for you?

It depends what framework you use, but in Remix I do that: https://github.com/rphlmr/supa-fly-stack/blob/3b4bf3e83e79a39bf1af23b222b2adc964a850fd/app/integrations/supabase/client.ts

Basically: exporting an already instantiated supabase client (since node only load imports once).

**Comment by hf** [comment]
This issue was the exact reason we added the warning. Thanks @rphlmr!

**Comment by maaajo** [comment]
@rphlmr Thanks! I was able to find the culprit in my case. It's the Strict Mode running component twice thus causing supabase client to be created twice. When I disable strict mode the warning is gone and there is only one instance

**Comment by jstdk** [comment]
@hf, my app is connected to two Supabase instances, which I create separately. In this scenario, how can I prevent this warning?

**Comment by sh41** [comment]
> [@hf](https://github.com/hf), my app is connected to two Supabase instances, which I create separately. In this scenario, how can I prevent this warning?

I had a similar issue, which I've resolved in #1100. The PR allows me to specify a different storage key for each client, which doesn't result in a warning. If multiple clients with the same storage key are instantiated the warning will still be seen.
